### PR TITLE
Apply non-null migration for hinted files

### PR DESCRIPTION
**Pull request by cbracken**
This applies non-null by default (NNBD) migration for files with hints.
All remaining non-migrated files are marked `// @dart = 2.9`.

Partial fix for #606.

### CODE REVIEW

File: lib/src/async/concat.dart [hunk 1]
```diff
@@ -40,15 +40,15 @@ class _ConcatStream<T> extends Stream<T> {
...
     cancelOnError = true == cancelOnError;
-    StreamSubscription<T>/*?*/ currentSubscription;
-    StreamController<T>/*?*/ controller;
+    StreamSubscription<T>? currentSubscription;
+    StreamController<T>? controller;
     final iterator = _streams.iterator;
 
     void nextStream(StreamController<T> controller) {
-      /*late*/ bool hasNext;
+      late bool hasNext;
```
Comment (yjbanov) [inline lib/src/async/concat.dart:51]: `final`?

File: test/collection/delegates/iterable_test.dart [hunk 2]
```diff
@@ -97,15 +98,13 @@ void main() {
 
     test('forEach', () {
       final it = delegatingIterable.iterator;
-      expect(it.current, isNull);
```
Comment (yjbanov) [inline test/collection/delegates/iterable_test.dart:100]: Should this `expect` `throws` instead?

File: test/collection/delegates/iterable_test.dart [hunk 3]
```diff
@@ -97,15 +98,13 @@ void main() {
...
       expect(it.moveNext(), isTrue);
       expect(it.current, equals('a'));
       expect(it.moveNext(), isTrue);
       expect(it.current, equals('b'));
       expect(it.moveNext(), isTrue);
       expect(it.current, equals('cc'));
       expect(it.moveNext(), isFalse);
-      expect(it.current, isNull);
```
Comment (yjbanov) [inline test/collection/delegates/iterable_test.dart:108]: Should this `expect` `throws` instead?

**Review by yjbanov — APPROVED** [review]
lgtm

File: test/collection/delegates/iterable_test.dart (hunk 2 above)
Comment (cbracken) [reply test/collection/delegates/iterable_test.dart:100]: Oh weird -- I thought I removed those two...

Comment (cbracken) [reply test/collection/delegates/iterable_test.dart:100]: Ah I did. Review race condition.

File: test/collection/delegates/iterable_test.dart (hunk 3 above)
Comment (cbracken) [reply test/collection/delegates/iterable_test.dart:108]: Done.

File: lib/src/async/concat.dart (hunk 1 above)
Comment (cbracken) [reply lib/src/async/concat.dart:51]: Done.

**Pull request merged**